boot/freeldr/freeldr/arch/uefi/CMakeLists.txt

```diff
@@ -0,0 +1,210 @@
+
+include_directories(BEFORE
+    ${REACTOS_SOURCE_DIR}/boot/environ/include/efi
+    ${REACTOS_SOURCE_DIR}/boot/freeldr/freeldr
+    ${REACTOS_SOURCE_DIR}/boot/freeldr/freeldr/include
+    ${REACTOS_SOURCE_DIR}/boot/freeldr/freeldr/arch/uefi/include)
+
+include_directories(
+    ${REACTOS_SOURCE_DIR}/ntoskrnl/include
+    ${REACTOS_SOURCE_DIR}/sdk/lib/cmlib
+    ${REACTOS_SOURCE_DIR}/sdk/include/reactos/libs
+    ${REACTOS_SOURCE_DIR}/sdk/include/reactos/elf
+    ${REACTOS_SOURCE_DIR}/sdk/include/reactos/uefi)
+
+add_definitions(-DUEFIBOOT -D_NTHAL_ -D_BLDR_ -D_NTSYSTEM_)
+
+list(APPEND FREELDR_BOOTLIB_SOURCE
+    ../../lib/arcsupp.c
+    ../../lib/debug.c
+    ../../lib/peloader.c
+    ../../lib/cache/blocklist.c
+    ../../lib/cache/cache.c
+    ../../lib/comm/rs232.c
+    ../../lib/fs/btrfs.c
+    ../../lib/fs/ext2.c
+    ../../lib/fs/fat.c
+    ../../lib/fs/fs.c
+    ../../lib/fs/iso.c
+    ../../lib/fs/ntfs.c
+    ../../lib/inifile/ini_init.c
+    ../../lib/inifile/inifile.c
+    ../../lib/inifile/parse.c
+    ../../lib/mm/meminit.c
+    ../../lib/mm/mm.c
+    ../../lib/mm/heap.c)
+
+list(APPEND FREELDR_NTLDR_SOURCE
+    ${REACTOS_SOURCE_DIR}/ntoskrnl/config/cmboot.c
+    ../../ntldr/conversion.c
+    ../../ntldr/registry.c
+    ../../ntldr/winldr.c
+    ../../ntldr/wlmemory.c
+    ../../ntldr/wlregistry.c)
+
+list(APPEND FREELDR_ARC_SOURCE
+    ../../arcname.c
+    ../../arch/drivers/hwide.c
+    ../../arch/arcemul.c
+    ../../arch/archwsup.c
+    ../../disk/disk.c
+    ../../disk/partition.c
+    ../../disk/ramdisk.c)
+
+list(APPEND FREELDR_BOOTMGR_SOURCE
+    ../../include/freeldr.h
+    ../../cmdline.c
+    ../../custom.c
+    ../../linuxboot.c
+    ../../miscboot.c
+    ../../options.c
+    ../../oslist.c
+    ../../ui/directui.c
+    ../../ui/minitui.c
+    ../../ui/noui.c
+    ../../ui/tui.c
+    ../../ui/tuimenu.c
+    ../../ui/ui.c
+    ../../ui/video.c)
+
+list(APPEND FREELDR_ARC_SOURCE
+    ../../lib/fs/pxe.c
+    stubs.c
+    uefiutil.c
+    ueficon.c
+    uefisetup.c
+    uefivid.c)
+
+list(APPEND FREELDR_BASE_ASM_SOURCE)
+
+if(ARCH STREQUAL "i386")
+    list(APPEND FREELDR_BOOTMGR_SOURCE
+        ../../arch/i386/i386bug.c)
+    list(APPEND FREELDR_BASE_ASM_SOURCE
+        ../../arch/i386/multiboot.S)
+
+    list(APPEND FREELDR_COMMON_ASM_SOURCE
+        ../../arch/i386/drvmap.S
```

Comment:
As for the other 3 files, those `../../arch/` can be replaced by just `../`.

Unless sometime in the near future you find the way how to share that list of files with the original freeldr's cmakelists, and avoid the copypaste. As done by e.g. ntoskrnl / ntkrnlmp.